includes/csm-mode.php

```diff
@@ -0,0 +1,116 @@
+<?php
+// Exit if accessed directly
+if(!defined('ABSPATH')){
+	exit;
+}
+
+/**
+ * Enable Coming Soon/Maintenance Mode
+ */
+class Fiber_Admin_CSM_Mode{
+	private $mode = '';
+	
+	public function __construct(){
+		// Enable Coming Soon/Maintenance Mode
+		$this->mode = fiad_get_csm_mode_option('mode');
+		$this->fiad_create_template_if_not_exists();
+		add_action('template_redirect', [$this, 'fiad_preview_csm_page']);
+		if(fiad_get_csm_mode_option('enable')){
+			add_filter('template_include', [$this, 'fiad_csm_content']);
+			add_action('wp_head', [$this, 'fiad_csm_extra_css']);
+			add_action('wp_footer', [$this, 'fiad_csm_extra_js']);
+		}
+	}
+	
+	//No Header & Footer Page
+	public function fiad_csm_content($template){
+		if(!current_user_can('edit_themes') || !is_user_logged_in()){
+			$new_template = dirname(__FILE__) . '/templates/' . $this->mode . '.php';
+			if($new_template){
+				return $new_template;
+			}
+		}
+		
+		return $template;
+	}
+	
+	public function fiad_preview_csm_page(){
+		$preview_mode = ev_array_key_exists('preview', $_GET);
+		if($preview_mode){
+			add_filter('template_include', function($template){
```

Comment:
Why don't we use `add_filter` in `construct`? I don't think it's a good idea to put it here since the theme must have possibility to override this filter.